components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -267,6 +267,15 @@ public OAuth2IntrospectionResponseDTO buildIntrospectionResponse(OAuth2TokenVali
             }
         }
 
+        // Adding the AccessTokenDO as a context property for further use
+        AccessTokenDO accessTokenDO;
+        try {
+            accessTokenDO = OAuth2Util.findAccessToken(validationRequest.getAccessToken().getIdentifier(), false);
```

Comment:
can simplify `validationRequest.getAccessToken().getIdentifier(`) as `oAuth2Token.getIdentifier()`

Reply:
also, let's set the second param in the method to true to include expired tokens as the token expiry validations are later handled in this method

Reply:
Added requested change by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1916/commits/d1efcb0613ea33ad628abe8d13fbbe5ae75e0327